profiling/src/exception.rs

```diff
@@ -45,12 +45,31 @@ impl ExceptionProfilingStats {
         self.next_sample = self.poisson.sample(&mut self.rng) as u32;
     }
 
-    fn track_exception(&mut self, name: String) {
+    fn track_exception(
+        &mut self,
+        #[cfg(php7)] exception: *mut zend::zval,
+        #[cfg(php8)] exception: *mut zend::zend_object,
+    ) {
         if let Some(next_sample) = self.next_sample.checked_sub(1) {
             self.next_sample = next_sample;
             return;
         }
 
+        #[cfg(php7)]
+        let exception_name = unsafe { (*(*exception).value.obj).class_name() };
+        #[cfg(php8)]
+        let exception_name = unsafe { (*exception).class_name() };
+
+        #[cfg(php7)]
+        let message = unsafe {
+            zend::zai_str_from_zstr(zend::zai_exception_message((*exception).value.obj).as_mut())
+                .into_string()
+        };
+        #[cfg(php8)]
+        let message = unsafe {
+            zend::zai_str_from_zstr(zend::zai_exception_message(exception).as_mut()).into_string()
+        };
```

Comment:
Since the duplicated expression isn't simple, I'd prefer to deduplicate it like this. What do you think?

```suggestion
        let message = unsafe {
            #[cfg(php7)]
            let exception_obj = (*exception).value.obj;
            #[cfg(php8)]
            let exception_obj = exception;
            zend::zai_str_from_zstr(zend::zai_exception_message(exception_obj).as_mut())
                .into_string()
        };
```

Reply:
Looks good!